server/src/main/java/org/apache/druid/catalog/model/CatalogUtils.java

```diff
@@ -68,12 +68,20 @@ public static Granularity asDruidGranularity(String value)
     if (Strings.isNullOrEmpty(value) || value.equalsIgnoreCase(DatasourceDefn.ALL_GRANULARITY)) {
       return Granularities.ALL;
     }
+    Granularity granularity;
     try {
-      return new PeriodGranularity(new Period(value), null, null);
+      granularity = Granularity.fromString(value);
     }
     catch (IllegalArgumentException e) {
-      throw new IAE(StringUtils.format("'%s' is an invalid period string", value));
+      try {
+        granularity = new PeriodGranularity(new Period(value), null, null);
+      }
+      catch (IllegalArgumentException e2) {
+        throw new IAE(StringUtils.format("[%s] is an invalid granularity string", value));
```

Comment:
`DruidException` instead of `IAE`:
```suggestion
       throw InvalidInput.exception("[%s] is an invalid granularity string.", value);
```

Reply:
fixed

Reply:
This caused a test failure. i see that there are catch blocks in places expecting IAE, dont want to fuss with that at the moment. I think we can change all of these throws in a different pr. ok with you?

Reply:
Yep, sounds good to me. I noticed that we throw `IAE` in a bunch of places in `CatalogUtils` which can be addressed in a separate PR.